homeassistant/components/tado/device_tracker.py

```diff
@@ -99,33 +93,15 @@ async def _async_update_info(self):
         Returns boolean if scanning successful.
         """
         _LOGGER.debug("Requesting Tado")
-
-        if self.websession is None:
-            self.websession = async_create_clientsession(
-                self.hass, cookie_jar=aiohttp.CookieJar(unsafe=True)
-            )
-
         last_results = []
 
-        try:
-            async with asyncio.timeout(10):
-                # Format the URL here, so we can log the template URL if
-                # anything goes wrong without exposing username and password.
-                url = self.tadoapiurl.format(
-                    home_id=self.home_id, username=self.username, password=self.password
-                )
-
-                response = await self.websession.get(url)
-
-                if response.status != HTTPStatus.OK:
-                    _LOGGER.warning("Error %d on %s", response.status, self.tadoapiurl)
-                    return False
-
-                tado_json = await response.json()
-
-        except (asyncio.TimeoutError, aiohttp.ClientError):
-            _LOGGER.error("Cannot load Tado data")
-            return False
+        await self.hass.async_add_executor_job(self.tadoconnector.setup)
+        if self.home_id is None:
+            tado_json = await self.hass.async_add_executor_job(self.tadoconnector.getMe)
```

Comment:
The `getMe` method is already called in `TadoConnector.setup`. We should refactor further to deduplicate those calls.

Reply:
No longer relevant, whereas the `ScannerEntity` is being implemented.